### PR TITLE
Feature/1283 conversion project stores directive academy order issued

### PR DESCRIPTION
**Pull request by lozette**
## Changes

_Add a summary of the changes in this pull request_

## Checklist

- [ ] Attach this pull request to the appropriate card in Trello.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

File: db/migrate/20230317114940_add_directive_academy_order_to_projects.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class AddDirectiveAcademyOrderToProjects < ActiveRecord::Migration[7.0]
+  def up
+    add_column :projects, :directive_academy_order, :boolean, default: false
+
+    Project.all.each do |project|
+      project.directive_academy_order = !(project.task_list_type == "Conversion::Voluntary::TaskList")
```
Comment (lozette) [inline db/migrate/20230317114940_add_directive_academy_order_to_projects.rb:6]: I actually hate this syntax but the linter is demanding it. I think it's quite obfuscated.

**Comment by lozette** [comment]
[Conversion project stores 'directive academy order issued?'](https://trello.com/c/RRw19HkF/1283-conversion-project-stores-directive-academy-order-issued)

File: app/forms/conversion/create_project_form.rb [hunk 2]
```diff
@@ -36,11 +38,17 @@ class NegativeValueError < StandardError; end
   validate :establishment_exists, if: -> { urn.present? }
   validate :trust_exists, if: -> { incoming_trust_ukprn.present? }
 
+  validates :directive_academy_order, inclusion: {in: ["true", "false"]}
```
Comment (lozette) [inline app/forms/conversion/create_project_form.rb:41]: These are strings not booleans, as they're coming from a form I think? Bools didn't work here. Happy to be corrected if I've done it wrong!

File: app/forms/conversion/create_project_form.rb [hunk 3]
```diff
@@ -36,11 +38,17 @@ class NegativeValueError < StandardError; end
   validate :establishment_exists, if: -> { urn.present? }
   validate :trust_exists, if: -> { incoming_trust_ukprn.present? }
 
+  validates :directive_academy_order, inclusion: {in: [true, "true", false, "false"]}
```
Comment (lozette) [inline app/forms/conversion/create_project_form.rb:41]: This is awful but it seems like these are bools or strings depending of whether they're coming from a test or the UI?!

File: db/migrate/20230317114940_add_directive_academy_order_to_projects.rb (hunk 1 above)
Comment (mec) [reply db/migrate/20230317114940_add_directive_academy_order_to_projects.rb:6]: Yeah, it's pretty hard to parse for me too! The only thing I could propose would be:

```
def directive_academy_required(project)
return false if project.task_list_type == "Conversion::Voluntary::TaskList"
return true if project.task_list_type == "Conversion::Inoluntary::TaskList"
end
```

File: app/forms/conversion/create_project_form.rb (hunk 3 above)
Comment (mec) [reply app/forms/conversion/create_project_form.rb:41]: I feel like these should be strings in the tests as that is what gets submitted from the form - but it is not a deal breaker - we have the same thing for `assigned_to_regional_caseworker_team`

**Review by mec — APPROVED** [review]
I left a couple of comments - nothing that would stop this going in though. 👍
